litmus-portal/graphql-server/pkg/k8s/cluster.go

```diff
@@ -2,94 +2,74 @@ package k8s
 
 import (
 	"context"
-	appsv1 "k8s.io/api/apps/v1"
-	apiv1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
 	"log"
 	"os"
 	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/restmapper"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	memory "k8s.io/client-go/discovery/cached"
+)
+
+var (
+	decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	dr              dynamic.ResourceInterface
+	AgentNamespace  = os.Getenv("AGENT_NAMESPACE")
 )
 
-func CreateDeployment(namespace, token string) (*appsv1.Deployment, error) {
-	deployerImage := os.Getenv("DEPLOYER_IMAGE")
-	subscriberSC := os.Getenv("AGENT_SCOPE")
-	selfDeployerSvcAccount := "self-deployer-namespace-account"
-	if subscriberSC == "cluster" {
-		selfDeployerSvcAccount = "self-deployer-admin-account"
+// This function handles cluster operations
+func ClusterOperations(manifest string, namespace string) (*unstructured.Unstructured, error) {
```

Comment:
Better to name the function something like `CreateResource`?

Reply:
This can be change